create_jenkins_job.py

```diff
@@ -217,6 +217,11 @@ def create_job(os_name, job_name, template_file, additional_dict):
             'cmake_build_type': 'None',
         })
 
+        create_job(os_name, 'test_ci_recovery_' + os_name, 'ci_job.xml.em', {
+            'cmake_build_type': 'None',
+            'with_eof_recovery': True
+        })
```

Comment:
As mentioned in https://github.com/ros2/ci/pull/703#issuecomment-1558027258 I'm in favor of hand-hacking `test_ci_linux` rather than creating a one-time job for testing.